examples/run/run.cpp

```diff
@@ -29,6 +29,7 @@
 #include "json.hpp"
 #include "linenoise.cpp/linenoise.h"
 #include "llama-cpp.h"
+#include "llama-impl.h"
```

Comment:
We cannot include implementation files in the user code.

Reply:
Should I move this code to somewhere else like common.cpp or leave it duplicated like prior to this PR?

Reply:
AFAIK we already had `string_format` in `common` so you can use it.

Regarding project structure, think of `src/*` as the kernel, `include/*` as the syscall and everything under `common/*` and `examples/*` is user-space code. In this case, the format function is implemented both in src and common because it's not a public API, so it's fine to duplicate it.

Reply:
Changed to use common.cpp version